### PR TITLE
Fix/normal hashing

### PR DESCRIPTION
**Pull request by jsburklund**
Fix an issue with the hashing function that didn't work properly when theta=Pi. Also explicitly used rounding functions to allocate the correct amount of memory, as well as the correct hashed index.

### CODE REVIEW

File: pointmatcher/DataPointsFilters/NormalSpace.cpp [hunk 1]
```diff
@@ -84,29 +84,33 @@ void NormalSpaceDataPointsFilter<T>::inPlaceFilter(DataPoints& cloud)
...
-		const T theta = std::acos(normals(2, i)); 
-		//Phi = azimuthal angle in [0 ; 2pi] 
+		const T theta = std::acos(normals(2, i));
+		//Phi = azimuthal angle in [0 ; 2pi]
 		const T phi = std::fmod(std::atan2(normals(1, i), normals(0, i)) + 2. * M_PI, 2. * M_PI);
-		
-		//assert(theta >= 0. and theta =< M_PI and phi >= 0. and phi <= 2.*M_PI);
-		
+
+		// Catch normal space hasing errors
```
Comment (remod) [inline pointmatcher/DataPointsFilters/NormalSpace.cpp]: Spelling

**Comment by remod** [comment]
Are all the assertions after merging this PR fulfulled?

File: pointmatcher/DataPointsFilters/NormalSpace.cpp [hunk 2]
```diff
@@ -172,7 +176,12 @@ template <typename T>
 std::size_t NormalSpaceDataPointsFilter<T>::bucketIdx(T theta, T phi) const
 {
 	//Theta = polar angle in [0 ; pi] and Phi = azimuthal angle in [0 ; 2pi]
-	return static_cast<std::size_t>(theta / epsilon) * static_cast<std::size_t>(2. * M_PI / epsilon) + static_cast<std::size_t>(phi / epsilon);
+	assert((theta >= 0.0) && (theta <= static_cast<T>(M_PI)) && (phi >= 0) && (phi <= 2.0*((float) static_cast<T>(M_PI))));
```
Comment (remod) [inline pointmatcher/DataPointsFilters/NormalSpace.cpp:179]: Sorry, I only noticed it now: you could split the assertion up in multiple assertions, and add output if they fail: https://stackoverflow.com/questions/3692954/add-custom-messages-in-assert

You can still merge it, only if you want to change it.

**Review by remod — APPROVED** [review]

**Review by YoshuaNava — APPROVED** [review]

**Comment by jsburklund** [comment]
ready to merge. May want to add commenting to assertions.

**Pull request merged**